dig.go

```diff
@@ -128,6 +129,26 @@ func Group(group string) ProvideOption {
 	})
 }
 
+// ConstructorInfo provides information about the constructor's inputs and outputs
+// types as strings, as well as the ID of the constructor supplied to the Container.
+//
+// ID is a unique integer representing the constructor node in the dependency graph.
+// Inputs is a string slice with the types of the input parameters of the constructor.
+// Outputs is a string slice with the types of the results produced by the constructor.
+type ConstructorInfo struct {
+	ID      int
+	Inputs  []string
```

Comment:
Hmm.. That's a good point. The main rationale behind exposing this as strings was the usage in `fx` but I can see keeping backwards compatibility becoming a mess if we want to expose more things in the future. 

What if I add a `ConstructorInfoOption`s and expose that as a parameter to `NewConstructorInfo`?

```go
func NewConstructorInfo(opts ...ConstructorInfoOption) ConstructorInfo
```

That way in the future if we want to expose not only the names of the types of the parameters/returns but the type value themselves, or `Group`/`Name`, user can construct the ConstructorInfo with whatever options they want to get out from `WithInfo`:

```go
var name string
var group string
var paramType []reflect.Type

info := NewConstructorInfo(WithNameInfo(&name), WithGroupInfo(&group), WithParamTypeInfo(&paramType))
c.Provide(ctor, WithInfo(&info))
```

Reply:
Yeah, I'm reluctant to expose just the strings as well. What do you think about opaque `Input` and `Output` types that happen to implement `fmt.Stringer` (`String() string`), reporting the current strings, but leave us room for exposing more information in the future?

```
type ConstructorInfo struct {
  ID int
  Inputs []*Input
  Outputs []*Output
}

type Input struct{ /* contains unexported fields */ }

func (*Input) String() string
```

We might even make `ConstructorInfo` completely opaque for now and add `Inputs() []*Input` and `Outputs() []*Output` methods if we're unsure, but that one is easier to mess with already since it's the "entry" point here.